compiler-rt/lib/radsan/tests/radsan_test_interceptors.cpp

```diff
@@ -134,18 +134,35 @@ TEST(TestRadsanInterceptors, openDiesWhenRealtime) {
   auto func = []() { open("./file.txt", O_RDONLY); };
   expectRealtimeDeath(func, "open");
   expectNonrealtimeSurvival(func);
+  std::remove("./file.txt");
 }
 
 TEST(TestRadsanInterceptors, openatDiesWhenRealtime) {
   auto func = []() { openat(0, "./file.txt", O_RDONLY); };
   expectRealtimeDeath(func, "openat");
   expectNonrealtimeSurvival(func);
+  std::remove("./file.txt");
 }
 
 TEST(TestRadsanInterceptors, creatDiesWhenRealtime) {
-  auto func = []() { creat("./file.txt", O_TRUNC); };
+
+  /*
+    The creat function doesn't seem to work on an ubuntu image when the path is
+    in a Docker shared volume. For now, to keep testing convenient with a local
+    Docker container, we just put it somewhere that's not in the shared volume
+    (/tmp). This is volatile and will be cleaned up as soon as the container is
+    stopped. */
+
+#if SANITIZER_LINUX
+  auto const filepath = "/tmp/file.txt";
```

Comment:
If we are throwing something in /tmp/, want to make this file something super specific that can't possibly overwrite something the user may have there?

aka

`/tmp/radsan_test_file.txt`

If it isn't hard, you could also add some random UUID or something to make 100% sure this isn't destructive to the user

Reply:
Yeah, good point! Really, there's too much duplication of "file.txt" everywhere in the tests anyway (we were lazy!), so I'll refactor it to be a function that returns "/tmp/radsan_test_file.txt" on Linux and "./radsan_test_file.txt" on macOS.